.github/workflows/windows.yml

```diff
@@ -55,16 +84,30 @@ jobs:
             arch: x64
             vs-release-dirname: x64
     steps:
-      - uses:  actions/checkout@v2
-      - name:  Install packages
+      - name:  Prepare vcpkg for cache restore
+        id:    prep-vcpkg
+        shell: pwsh
+        run: |
+          mv c:\vcpkg c:\vcpkg-bak
+          md c:\vcpkg -ea 0
+
+      - name: Restore the most recent cache of vcpkg
+        uses: actions/cache@v2
+        with:
+          path: c:\vcpkg
+          key: vcpkg-${{ matrix.conf.arch }}-
```

Comment:
Something seems missing in this line?

Reply:
It sure looks like it, but this "danging" key actually gives us a lot of very interesting benefits.

What this syntax says is "_give me to the most recent successfully generated cache that matches the pattern_" **vcpkg-${{ matrix.conf.arch }}-\***

Where as further up in the debug section, our cache key includes the week-number timestamp:

**vcpkg-${{ matrix.conf.arch }}-YYYYWW**


Beacuse the release builds `needs:` the debug build, and the debug build key demands the cache be current as-of-this week, then by extension our release job will only ever use the latest weekly generated cache (and nothing older).

Even the faiilure scenario I described below.  Say we do have to manually trigger a cache regen mid-week by bumping the key from `-1` to `-2`.  The debug build would run, create a fresh cache, and because of this pattern and "newest" search, the release build would use the `-2` cache because it's simply newer.

The other benefit **vcpkg-${{ matrix.conf.arch }}-\*** gives us is that it's avoid those time-boundaries that have broken our cache before (one job finishes 2 seconds before midnight on a sunday, the next job start 2 days after midnight and is given an entirely new month and week value!).  This avoids that.. the release build will use the successful cache most-recent-in-time, regardless of date boundaries.